src/hooks/useFauxCaretCssVars.ts

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import Path from '../@types/Path'
+import { isSafari, isTouch } from '../browser'
+import { isEndOfElementNode, isStartOfElementNode } from '../device/selection'
+import editingValueStore from '../stores/editingValue'
+
+type FauxCaretType = 'none' | 'thoughtStart' | 'thoughtEnd' | 'noteStart' | 'noteEnd'
+
+/** Returns CSS variables that will suppress faux carets at the start or end of thoughts or notes.
+ * */
+const useFauxCaretCssVars = (
+  editing: boolean | null,
+  // Undo followed by redo causes the component to re-mount with the same prop values,
+  // but fadeThoughRef is re-created and will trigger an update.
+  fadeThoughtElement: HTMLDivElement | null,
+  isCursor: boolean,
+  isTableCol1: boolean,
+  path: Path,
+) => {
+  const [fauxCaretType, setFauxCaretType] = useState<FauxCaretType>('none')
+  const noteFocus = useSelector(state => state.noteFocus)
+
+  // Hide the faux caret when typing occurs.
+  editingValueStore.useEffect(() => {
+    if (isTouch && isSafari()) setFauxCaretType('none')
+  })
+
+  // If the thought isCursor and edit mode is on, position the faux cursor at the point where the
+  // selection is created.
+  useEffect(() => {
+    if (!isTouch || !isSafari()) return
+    if (editing && isCursor) {
+      // The selection ranges aren't updated until the end of the frame when the thought is focused.
+      setTimeout(() => {
+        if (editing && isCursor) {
```

Comment:
I *think* `editing` and `isCursor` will always be true here, as the setTimeout callback has closure over the props. `editing` and `isCursor` in the outer scope will update since they are part of the memo dependencies, but I don't believe that will extend to the `setTimeout` callback.

If you need to ensure that `editing` and `isCursor` are still true after the `setTimeout` callback has fired, then you will need to import the `store` and do a fresh `store.getState()`.